converter/datamodel.py

```diff
@@ -174,8 +192,14 @@ def from_magetab(cls, le_attributes, extract_attributes, assay_attributes):
         return cls(alias, accession, technology_type[0], protocolrefs, sampleref, label, array_design[0])
 
     @classmethod
-    def from_json(self):
-        pass
+    def from_dict(cls, assay_dict):
```

Comment:
Couldn't you have maybe some object factory for cls objects that receive the dictionary and a type of cls, and then pick the relevant fields for each type of cls. At least then you would avoid re-writing all the common fields being picked up.

Reply:
or maybe some inheritance between cls object types?